### PR TITLE
multiple-pause-resume: change default values and fix serious loop index bug

### PR DESCRIPTION
**Pull request by fredoh9**
@ranj063 reported this problem. Occasionally it timeout too early.
```
2021-06-30 17:20:20 UTC Sub-Test: [INFO] pipeline: Port1 with aplay
2021-06-30 17:20:20 UTC Sub-Test: [INFO] pipeline: Port2 with aplay
2021-06-30 17:20:20 UTC Sub-Test: [INFO] Check expect exit status
2021-06-30 17:20:20 UTC Sub-Test: [INFO] Port1 to command: aplay -D hw:0,1 -r 48000 -c 2 -f S16_LE -vv -i /dev/zero -q
spawn aplay -D hw:0,1 -r 48000 -c 2 -f S16_LE -vv -i /dev/zero -q^M
2021-06-30 17:20:20 UTC Sub-Test: [INFO] Port2 to command: arecord -D hw:0,2 -r 48000 -c 2 -f S16_LE -vv -i /dev/null -q
2021-06-30 17:20:20 UTC Sub-Test: [INFO] wait for expect process finished
===> Fred: only after 5 second, it timeout for 250 second!
2021-06-30 17:20:25 UTC Sub-Test: [ERROR] Still have expect process not finished after wait for 250
root        7937    7931 34 17:20 pts/1    00:00:01 aplay -D hw:0,1 -r 48000 -c 2 -f S16_LE -vv -i /dev/zero -q
root        7946    7942 36 17:20 pts/2    00:00:01 arecord -D hw:0,2 -r 48000 -c 2 -f S16_LE -vv -i /dev/null -q
root        7962    6577  0 17:20 pts/0    00:00:00 grep -E aplay|arecord
2021-06-30 17:20:25 UTC Sub-Test: [ERROR] Starting func_exit_handler(), exit status=1, FUNCNAME stack:
2021-06-30 17:20:25 UTC Sub-Test: [ERROR]  main()  @  /home/ubuntu/sof-test/test-case/multiple-pause-resume.sh
```

### CODE REVIEW

File: test-case/multiple-pause-resume.sh [hunk 1]
```diff
@@ -36,10 +36,10 @@ OPT_NAME['r']='loop'     OPT_DESC['r']='pause resume repeat count'
 OPT_HAS_ARG['r']=1         OPT_VAL['r']=3
 
 OPT_NAME['i']='min'      OPT_DESC['i']='random range min value, unit is ms'
-OPT_HAS_ARG['i']=1         OPT_VAL['i']='100'
+OPT_HAS_ARG['i']=1         OPT_VAL['i']='70'
 
 OPT_NAME['a']='max'      OPT_DESC['a']='random range max value, unit is ms'
-OPT_HAS_ARG['a']=1         OPT_VAL['a']='200'
+OPT_HAS_ARG['a']=1         OPT_VAL['a']='120'
```
Comment (marc-hb) [inline test-case/multiple-pause-resume.sh:42]: Could  you please fix the header above or the OPT_DESC or both to explain what this mysterious "range" is?

**Review by marc-hb — COMMENTED** [review]
The first commit is... wow.

**Review by ranj063 — APPROVED** [review]
Thanks for the quick fix @fredoh9

File: test-case/multiple-pause-resume.sh (hunk 1 above)
Comment (fredoh9) [reply test-case/multiple-pause-resume.sh:43]: I updated the description also

**Comment by marc-hb** [comment]
@fredoh9 could you quickly revert https://github.com/thesofproject/sof/commit/dcbee3cdf3dcd313d1642684ac54a6f980ce4e92 locally and run as detailed in https://github.com/thesofproject/sof/issues/3522#issuecomment-858859828 to make sure we don't have yet another "green failure" test?

File: test-case/multiple-pause-resume.sh (hunk 1 above)
Comment (keyonjie) [reply test-case/multiple-pause-resume.sh:43]: @fredoh9 usually I use "-i 20 -a 50 -l 100" locally to help to capture more issues within 2 hours, for daily, maybe "-i 20 -a 50 -l 20" could be good.

Comment (fredoh9) [reply test-case/multiple-pause-resume.sh:43]: @keyonjie do you recommend -i 20 -a 50 as default values?

Comment (keyonjie) [reply test-case/multiple-pause-resume.sh:43]: > @keyonjie do you recommend -i 20 -a 50 as default values?

yes, and run more loop (default 3 -> 20?) to capture issues more stably.

**Comment by fredoh9** [comment]
> @fredoh9 could you quickly revert [thesofproject/sof@dcbee3c](https://github.com/thesofproject/sof/commit/dcbee3cdf3dcd313d1642684ac54a6f980ce4e92) locally and run as detailed in [thesofproject/sof#3522 (comment)](https://github.com/thesofproject/sof/issues/3522#issuecomment-858859828) to make sure we don't have yet another "green failure" test?

I reverted it and I'm able to reproduce the problem with simply running './multiple-pause-resume.sh -r 25' this script. I tested on APL NOCODEC, I had to find old topology which has a MIXER.

File: test-case/multiple-pause-resume.sh (hunk 1 above)
Comment (fredoh9) [reply test-case/multiple-pause-resume.sh]: Thank you for your suggestion, I think good point. But default loop/repeat value should be carefully decided. Daily test use "/multiple-pause-resume.sh -r 25" if default loop is 20, this will take too much. If any one want to run more thorough test, they can specify that count with -l/-r options.
I slightly increased loop/round count.

File: test-case/multiple-pause-resume.sh [hunk 2]
```diff
@@ -139,7 +139,8 @@ exit 1
 END
 }
 
-max_wait_time=$[ 10 * $repeat_count ]
+# maximum 5 second per individual repeat
+max_wait_time=$[ 5 * $repeat_count ]
```
Comment (plbossart) [inline test-case/multiple-pause-resume.sh:144]: is this change related to the fix below? Seems not, so probably need to push the fix below to a quick PR, and the rest as improvements?

File: test-case/multiple-pause-resume.sh [hunk 3]
```diff
@@ -27,19 +27,20 @@ OPT_NAME['t']='tplg'     OPT_DESC['t']='tplg file, default value is env TPLG: $T
...
 OPT_NAME['c']='count'    OPT_DESC['c']='combine test pipeline count'
 OPT_HAS_ARG['c']=1         OPT_VAL['c']=2
 
 OPT_NAME['r']='loop'     OPT_DESC['r']='pause resume repeat count'
-OPT_HAS_ARG['r']=1         OPT_VAL['r']=3
+OPT_HAS_ARG['r']=1         OPT_VAL['r']=5
 
-OPT_NAME['i']='min'      OPT_DESC['i']='random range min value, unit is ms'
-OPT_HAS_ARG['i']=1         OPT_VAL['i']='100'
+#pause/resume interval will be randomly decided based on below min/max
```
Comment (plbossart) [inline test-case/multiple-pause-resume.sh:38]: did you mean
"pause/resume interval will be a random value bounded by the min and max values below" 
?

File: test-case/run-all-tests.sh [hunk 4]
```diff
@@ -228,7 +228,7 @@ test_multiple-pipeline-capture()
 }
 test_multiple-pause-resume()
 {
-	"$mydir"/multiple-pause-resume.sh -r 25
+	"$mydir"/multiple-pause-resume.sh -l "$small_loop" -r 25
```
Comment (plbossart) [inline test-case/run-all-tests.sh:231]: not quite able to understand what you are trying to do?

what does 'faster transition' mean? a reduction in the min/max range? a reduction of both min and max?
And what is this small_loop doing here?

**Review by plbossart — CHANGES_REQUESTED** [review]
I have many questions.

File: test-case/multiple-pause-resume.sh (hunk 1 above)
Comment (keyonjie) [reply test-case/multiple-pause-resume.sh]: sounds good.

File: test-case/multiple-pause-resume.sh (hunk 2 above)
Comment (fredoh9) [reply test-case/multiple-pause-resume.sh:144]: I agree, I will make a separate PR for improvement

File: test-case/multiple-pause-resume.sh (hunk 3 above)
Comment (fredoh9) [reply test-case/multiple-pause-resume.sh:38]: thank you for your clarification

File: test-case/run-all-tests.sh (hunk 4 above)
Comment (fredoh9) [reply test-case/run-all-tests.sh:231]: I'm trying to add a parameter for loop count.

faster transition by reductions of both min and max. This is nothing but sleep time, bounded by min and max, between pause and resume.

**Review by aiChaoSONG — APPROVED** [review]

**Review by keqiaozhang — APPROVED** [review]

**Comment by aiChaoSONG** [comment]
Only a variable clarification, no logic change, let merge.
